platform/machine/do/flight.go

```diff
@@ -108,12 +111,14 @@ func (df *flight) NewCluster(rconf *platform.RuntimeConfig) (platform.Cluster, e
 }
 
 func (df *flight) Destroy() {
-	for _, keyID := range []int{df.sshKeyID, df.fakeSSHKeyID} {
-		if keyID == 0 {
-			continue
-		}
-		if err := df.api.DeleteKey(context.TODO(), keyID); err != nil {
-			plog.Errorf("Error deleting key %v: %v", keyID, err)
+	if df.keyAdded {
```

Comment:
Unneeded for DO, since `sshKeyID == 0` or `fakeSSHKeyID == 0` serve that purpose.